File: java/pe003.java
public class pe003
{
    public static long lpf (long num)
    {
        long tempnum = num;
        long ans = 0;
 
        for (long factor = 2; factor < num/2; factor++) 
        {
            while (tempnum % factor == 0) 
            {
                tempnum /= factor;
            }
            if (tempnum == 1) 
            {
                ans = factor;
                break;
            }
        }
        return ans;
    }
 
    public static void main (String[] args)
    {
        System.out.println (lpf(600851475143L));
    }
}

File: nim/pe003.nim
discard """
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
"""

proc factors(num: int64): seq[int64] =
  var temp = num
  var ans: seq[int64] = @[]
  var d: int64 = 2
  while temp > 1:
    if temp mod d == 0:
      ans.add(d)
      temp = temp div d
    else:
      d += 1
  return ans

let fs = factors(600851475143)
let ans = fs[len(fs)-1]
echo ans